build/loaders/hot-self-accept-loader.js

```diff
@@ -30,18 +36,21 @@ module.exports = function (content, sourceMap) {
   `, sourceMap)
 }
 
-const nextPagesDir = resolve(__dirname, '..', '..', '..', 'pages')
-
-function getRoute (loaderContext, options) {
-  const pagesDir = resolve(loaderContext.options.context, 'pages')
-  const { resourcePath } = loaderContext
-  const dir = [pagesDir, nextPagesDir]
-    .find((d) => resourcePath.indexOf(d) === 0)
-
+function getRoute (resourcePath: string, options: Options) {
   if (!options.extensions) {
     throw new Error('extensions is not provided to hot-self-accept-loader. Please upgrade all next-plugins to the latest version.')
   }
 
+  if (!options.include) {
+    throw new Error('include option is not provided to hot-self-accept-loader. Please upgrade all next-plugins to the latest version.')
+  }
+
+  const dir = options.include.find((d) => resourcePath.indexOf(d) === 0)
```

Comment:
do the `indexOf` outside of the predicate to a variable and use that in here

Reply:
How'd you mean 🤔